nav2_theta_star_planner/README.md

```diff
@@ -0,0 +1,92 @@
+- ` .w_heuristic_cost ` : it has been provided to have an admissible heuristic, so the recommendation would be to change its value only when required. Usually set is at the same value as `w_euc_cost` or 1.0 (whichever is lower), though you may increase the value of `w_heuristic_cost` to speed up the process.
+
+Below are the default values of the parameters :
+```
+planner_server:
+  ros__parameters:
+    planner_plugin_types: ["nav2_theta_star_planner/ThetaStarPlanner"]
+    use_sim_time: True
+    planner_plugin_ids: ["GridBased"]
+    GridBased:
+      how_many_corners: 8
+      w_euc_cost: 1.0
+      w_traversal_cost: 2.0
+      w_heuristic_cost: 1.0
+```
+Do note that the `global_costmap`'s `inflation_layer` values recommended for the above values of the parameter are - `cost_scaling_factor:4.0`, `inflation_radius: 5.5`
+
+## Usage Notes
+
+### Tuning the Parameters
+Before starting off, do note that the costmap_cost(curr,neigh) component after being operated (ie before being multiplied to its parameter and being substituted in g(init_rclcpp)) varies from 0 to 1.
+
+This planner uses the costs associated with each cell from the `global_costmap` as a measure of the point's proximity to the obstacles. Providing a gentle potential field that covers the entirety of the region (thus leading to only small pocket like regions of cost = 0) is recommended in order to achieve paths that pass through the middle of the spaces. A good starting point could be to set the `inflation_layer`'s parameters as - `cost_scaling_factor:10.0`, `inflation_radius: 5.5` and then to decrease the value of `cost_scaling_factor` to achieve the said potential field.
+
+Providing a gentle potential field over the region allows the planner to exchange the increase in path lengths for a decrease in the accumulated traversal cost, thus leading to an increase in the distance from the obstacles. This around a corner allows for naturally smoothing the turns and removes the requirement for an external path smoother.
+
+In order to achieve paths that stay in the middle of the spaces set `w_traversal_cost` at a higher value than `w_euc_cost`. To begin with, you can set the parameters to its default values and then increase the value of `w_traversal_cost` which, if required can be accompanied with a decrease in `w_euc_cost` to allow for an increase in the path length thus letting the path to settle in the middle regions of the spaces with lower costs on the costmap. While tuning the planner's parameters you can also change the `inflation_layer`'s parameters (of the global costmap) to tune the behavior of the paths.
```

Comment:
Are there any draw backs of this worth mentioning? (speed, for instance?)

Reply:
can you elaborate a bit on this

Reply:
You mention pushing up / down some parameters to get certain behavior. Usually in tuning getting some behavior comes at the expense of something else (maybe smoothness, maybe compute time, etc). I wanted to know for any tuning advise to get `Behavior A`, what `Cost A` accompanies it (if any)

Reply:
Got it!